foyer-storage/src/device/direct_fs.rs

```diff
@@ -162,14 +163,22 @@ impl Dev for DirectFsDevice {
             region_size = self.region_size(),
         );
 
-        let file = self.file(region).clone();
+        let mut file = self.file(region).clone();
         asyncify_with_runtime(&self.inner.runtime, move || {
             #[cfg(target_family = "unix")]
             use std::os::unix::fs::FileExt;
             #[cfg(target_family = "windows")]
-            use std::os::windows::fs::FileExt;
+            let written = {
+                let original = file.stream_position()?;
+                file.seek(std::io::SeekFrom::Start(offset))?;
+                let written = file.write(buf.as_aligned())?;
+                file.seek(std::io::SeekFrom::Start(original))?;
+                written
+            };
```

Comment:
IIUC, functions provided by `std::os::windows::fs::FileExt` can be used here to do positioned read/write.

Manually doing seek + io may lead to concurrent problems.

https://doc.rust-lang.org/std/os/windows/fs/trait.FileExt.html

Reply:
Done also let me know if it's ok to use the allow_warning I used it for mut warning on unix since.